server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -135,7 +148,12 @@ public void postParse(DocumentParserContext context) throws IOException {
         } else {
             timeSeriesId = routingPathFields.buildHash().toBytesRef();
         }
-        context.doc().add(new SortedDocValuesField(fieldType().name(), timeSeriesId));
+
+        if (this.useDocValuesSkipper && this.indexCreatedVersion.onOrAfter(IndexVersions.TIME_SERIES_ID_DOC_VALUES_SPARSE_INDEX)) {
+            context.doc().add(SortedDocValuesField.indexedField(fieldType().name(), timeSeriesId));
```

Comment:
SortedDocValuesField#indexedField creates a new SortedDocValuesField using a private static FieldType that has the skip index enabled. It's the same as discussed for SortedNumericDocValuesField in https://github.com/elastic/elasticsearch/pull/122161#discussion_r1953020972